external-docs/docs/guides/csv-to-csvw-github-action.md

```diff
@@ -0,0 +1,73 @@
+
+Commiting a file to any other location within the repository will NOT trigger the action.
+
+#### Process committed files
+
+Once the files are committed, the action runs the `csvcubed build` and `csvcubed inspect` commands. The outputs produced by the build command are saved at the `out/` folder at the root of the repository. For convenience, the action keeps the same folder structure within the `csv/` folder for the outputs produced. For example, for the `my_data.csv` file at `csv/my_folder/my_data.csv`, the action stores the CSV-Ws at `out/my_folder/my_data/`. 
+
+After running the build command, the action runs the `csvcubed inspect` command on the metadata JSON file within the CSV-W outputs. The inspect command output will also be saved at the `out/` folder similar to the build command output above. For example, for the metadata JSON file at `out/my_folder/my_data/my_data.csv-metadata.json`, a log file called `inspect_output.txt` will be produced and stored at `out/my_folder/my_data/`. This file consists of the inspect command output.
+
+#### Commit generated CSV-Ws and logs to repository
+
+In order to maintain the history of the outputs produced and allow the user to revert changes to the outputs, the action commits the `out/` folder to the repository after running the build and inspect commands.
+
+#### Publish generated CSV-Ws and logs to GitHub artefacts
+
+The action then publishes the CSV-Ws and logs to GitHub artefacts. The user can download a zip file consisting of the CSV-Ws and inspect command output from the artefacts section within the GitHub action run. More information on how to download the artefacts is available in [Download GitHub Action Artefacts](https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artefacts).
+
+#### Publish generated CSV-Ws and logs to GitHub Pages
+
+The action also presents the user with the generated CSV-Ws and logs in a user-friendly GitHub page. In this page, the files in the `out/` folder are listed as downloadable links. The URL to access the GitHub page is provided in GitHub pages setting which is discussed in the [User Manual](#user-manual) section below.
+
+## Organising Inputs
+
+The CSV-to-CSV-W GitHub action expects the user to organise the inputs as per the below criteria.
+
+* The CSV files and their configuration JSONs need to be inside the `csv/` folder.
+  
+* If a CSV file has a configuration JSON, the filename of the configuration JSON needs to match the filename of the CSV file. For example, `my_data.CSV` and `my_data.csv.json`.
+
+## User Manual
+
+CSV-to-CSV-W GitHub action can be used easily through the GitHub web console. Users who are experienced with GitHub can also clone the repository to the local machine and then commit files using the [GitHub Desktop Client](https://desktop.github.com/) or the [GitHub Command Line Interface](https://cli.github.com/).
```

Comment:
This section could be simplified by removing all CLI/github desktop instructions.